Geometry/DTGeometryBuilder/plugins/dd4hep/DTGeometryValidate.cc

```diff
@@ -0,0 +1,364 @@
+    
+    auto const& shape = fwGeometry_.getShapePars(chId.rawId());
+    
+    if(!shape) {
+      LogVerbatim("DTGeometry") << "Failed to get shape of DT chamber with detid: "
+                                << chId.rawId();
+      continue;
+    }
+    
+    compareShape(it, shape);
+  }
+
+  makeHistograms("DT Chamber");
+}
+
+void 
+DTGeometryValidate::validateDTLayerGeometry() {
+
+  clearData();
+  
+  vector<double> wire_positions;
+
+  for(auto const& it : dtGeometry_->layers()) {
+    DTLayerId layerId = it->id();
+    GlobalPoint gp = it->surface().toGlobal(LocalPoint(0.0, 0.0, 0.0)); 
+     
+    const TGeoMatrix* matrix = fwGeometry_.getMatrix(layerId.rawId());
+ 
+    if (!matrix) {     
+        LogVerbatim("DTGeometry") << "Failed to get matrix of DT layer with detid: " 
+                                  << layerId.rawId();
+        continue;
+    }
+
+    compareTransform(gp, matrix);
+
+    auto const& shape = fwGeometry_.getShapePars(layerId.rawId());
+
+    if(!shape) {
+      LogVerbatim("DTGeometry") << "Failed to get shape of DT layer with detid: "
+                                << layerId.rawId();
+      continue;
+    }
+      
+    compareShape(it, shape);
+    
+    auto const& parameters = fwGeometry_.getParameters(layerId.rawId());
+      
+    if(parameters == nullptr) {
+      LogVerbatim("DTGeometry") << "Parameters empty for DT layer with detid: " 
+				<< layerId.rawId();
+      continue;
+    }
+           
+    double width = it->surface().bounds().width();
+    assert(width == parameters[6]); 
+
+    double thickness = it->surface().bounds().thickness();
+    assert(thickness == parameters[7]);
+
+    double length = it->surface().bounds().length();
+    assert(length == parameters[8]);
+
+    int firstChannel = it->specificTopology().firstChannel();
+    assert(firstChannel == parameters[3]);
+
+    int lastChannel = it->specificTopology().lastChannel();
+    int nChannels = parameters[5];
+    assert(nChannels == (lastChannel-firstChannel)+1);
+
+    for(int wireN = firstChannel; wireN - lastChannel <= 0; ++wireN) {
+      double localX1 = it->specificTopology().wirePosition(wireN);
+      double localX2 = (wireN -(firstChannel-1)-0.5f)*parameters[0] - nChannels/2.0f*parameters[0];
+      wire_positions.emplace_back(getDiff(localX1, localX2));
+    }
+  }
+  
+  makeHistogram("DT Layer Wire localX", wire_positions);
+  makeHistograms("DT Layer");
+}
+
+void    
+DTGeometryValidate::compareTransform(const GlobalPoint& gp,
+				     const TGeoMatrix* matrix)
+{
+  double local[3] = { 0.0, 0.0, 0.0 };  
+  double global[3];
+
+  matrix->LocalToMaster(local, global);
+
+  double distance = getDistance(GlobalPoint(global[0], global[1], global[2]), gp);
+  globalDistances_.push_back(distance);
+}
+
+void 
+DTGeometryValidate::compareShape(const GeomDet* det, const float* shape)
+{
+  float shapeTopWidth;
+  float shapeBottomWidth;
+  float shapeLength;
+  float shapeThickness;
+
+  if(shape[0] == 1) {
```

Comment:
```suggestion
  if(shape[0] == 1.) {
```

Reply:
To remind the reader that `shape` contains floats.

Reply:
Except that the first element is a shape indicator. It should be an enum.